### PR TITLE
tests, libvmi: move adjust and delete cloudinit function from tests/utils to libvmi

### PR DESCRIPTION
**Pull request by ShellyKa13**
### What this PR does
Before this PR:
Having cloud-init functions in tests/utils

After this PR:
Moved and adjusted the functions used to libvmi, adjusted code accordingly.
Functions:
AddCloudInitConfigDriveData -> WithCloudInitConfigDriveData, WithCloudInitConfigEncodedDriveData
AddUserData -> deleted (replaced with already existing one)
addCloudInitDiskAndVolume -> deleted

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/utils.go [hunk 1]
```diff
@@ -192,7 +191,8 @@ func AddEphemeralDisk(vmi *v1.VirtualMachineInstance, name string, bus v1.DiskBu
 // Deprecated: Use libvmi directly
 func NewRandomVMIWithEphemeralDiskAndUserdata(containerImage string, userData string) *v1.VirtualMachineInstance {
 	vmi := NewRandomVMIWithEphemeralDisk(containerImage)
-	AddUserData(vmi, "disk1", userData)
+	cloudInitNoCloudEncodedUserData := libvmi.WithCloudInitNoCloudEncodedUserData(userData)
```
[the remark this answers is not in the document]
Comment (ShellyKa13) [reply tests/utils.go:194]: Refactoring this function is not so straight forward, I'm doing small PRs

File: pkg/libvmi/cloudinit.go [hunk 2]
```diff
@@ -99,6 +99,30 @@ func WithCloudInitConfigDriveUserData(data string) Option {
 	}
 }
 
+// WithCloudInitConfigDriveData adds cloud-init config-drive data.
+func WithCloudInitConfigDriveData(userData, networkData string) Option {
```
[the remark this answers is not in the document]
Comment (ShellyKa13) [reply pkg/libvmi/cloudinit.go:103]: There is a cloudinit refactor PR: #12145 that I will rebase on and it will align with your request

**Comment by enp0s3** [comment]
@ShellyKa13 Thanks! Please rebase when possible
/approve

File: tests/libvmifact/factory.go [hunk 3]
```diff
@@ -49,16 +50,19 @@ func NewFedora(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
...
-		withNonEmptyUserData,
 		libvmi.WithResourceMemory(cirrosMemory()),
 	}
 	cirrosOpts = append(cirrosOpts, opts...)
-	return libvmi.New(cirrosOpts...)
+	vmi := libvmi.New(cirrosOpts...)
+
+	// Supplied with no user data, Cirros image takes 230s to allow login
+	if !cloudInitNoCloudExists(vmi.Spec.Volumes) {
+		withNonEmptyUserData := libvmi.WithCloudInitNoCloud(libvmici.WithNoCloudEncodedUserData("#!/bin/bash\necho hello\n"))
```
Comment (nunnatsa) [inline tests/libvmifact/factory.go:62]: this is defined twice with the same value. Do you want to make it a local variable?

Comment (nunnatsa) [reply tests/libvmifact/factory.go]: Please ignore. sorry

**Comment by nunnatsa** [comment]
/lgtm

Thanks @ShellyKa13 !

**Pull request merged**